File: api/auth/Entities/Adresse.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;


namespace GoodFood.Auth.Entities
{
    [Table("adresse")]
    public class Adresse
    {
        [Key]
        [Column("id")]
        public int id {get; set;}

        [Column("pays")]
        [Required]
        public string pays {get; set;}

        [Column("ville")]
        [Required]
        public string ville {get; set;}

        [Column("code_postal")]
        [Required]
        public string codePostal {get; set;}

        [Column("rue")]
        [Required]
        public string rue {get; set;}

        [Column("complement")]
        public string complement {get; set;}
    }
}

File: app/good-food/src/app/shared/models/adresse.ts

export interface IAdresse {
  id?: number;
  pays?: string;
  codePostal?: string;
  rue?: string;
  ville?: string;
  complement?: string | null;
}

export class Adresse implements IAdresse {
  id?: number;
  pays?: string;
  codePostal?: string;
  rue?: string;
  ville?: string;
  complement?: string | null;
  constructor(opt?: Partial<IAdresse>) {
    Object.assign(this, opt);
  }

  toString(): string{
    return `${this.rue}, ${this.codePostal} ${this.ville} - ${this.pays}`;
  }
}
